bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt

```diff
@@ -284,9 +290,23 @@ private constructor(
             this.debug = true
         }
 
+        private fun getInstanceVersion(): String {
```

Comment:
We probably can't yet use `Public.getInstance(client)` because we're just building the client we'd need to use in that call. Would it still be useful to move this into `Public.kt`, perhaps into a companion object, to have it in the same place as similar calls? We might be able to reuse some of its code there.

Reply:
I would even go a step further and introduce a new method class `Instance` and place all future methods that are part of https://docs.joinmastodon.org/methods/instance/ in there. Doing so, it becomes more clear to the library user where to find our methods. 

`Public.kt` is not listed as a method in the API documentation, therefore I would vote to get rid of it entirely. We could move all methods related to https://docs.joinmastodon.org/methods/instance/ into a separate `Instance` method class, and everything that relates to https://docs.joinmastodon.org/methods/search/ into a new `Search` method class.

In general, it would make sense to quickly review the method class structure we have and compare it to the onev in the API documentation. I have created an issue for this. I think there are more cases (did not check tought):

https://github.com/andregasser/bigbone/issues/94

What do you think?

Reply:
Sounds good. Naming of the method classes is a bit weird with all of them being plural so far, but I'll add that to #94.